src/app/home/home.component.ts

```diff
@@ -42,21 +45,34 @@ export class HomeComponent implements OnInit {
     });
   }
 
+  openSnackBar(message: string, action: string) {
+    this._snackBar.open(message, action, {
+      duration: 7000,
+    });
+  }
+
   getWeather(location: string) {
     this.isLoading = true;
     this.weatherService.get(location)
       .pipe(finalize(() => { this.isLoading = false; }))
-      .subscribe((res: IWeather) => {
-        this.weather = res;
-        this.current = this.weather.current;
-    });
+      .subscribe(
+        (res: IWeather) => {
+          this.weather = res;
+          this.current = this.weather.current;
+        },
+        (err: IError) => {
+          this.openSnackBar(err.error.error.message, 'Close');
```

Comment:
What happens if the error message is null? I would add a previous validation and a default message in case any of the fields (`err`, `error`, `error` or `message`) is undefined.

Reply:
Condition added! Thanks!